widgetsnbextension/src/manager.js

```diff
@@ -219,45 +235,18 @@ export class WidgetManager extends ManagerBase {
    */
   _init_actions() {
     var notifier = Jupyter.notification_area.widget('widgets');
-    this.saveWidgetsAction = {
-      handler: function() {
-        this.get_state({
-          drop_defaults: true
-        }).then(function(state) {
-          Jupyter.notebook.metadata.widgets = {
-            'application/vnd.jupyter.widget-state+json': state
-          };
-          Jupyter.menubar.actions
-            .get('jupyter-notebook:save-notebook')
-            .handler({
-              notebook: Jupyter.notebook
-            });
-        });
-      }.bind(this),
-      icon: 'fa-truck',
-      help:
-        'Save the notebook with the widget state information for static rendering'
-    };
-    Jupyter.menubar.actions.register(
-      this.saveWidgetsAction,
-      'save-with-widgets',
-      'widgets'
-    );
-
-    this.clearWidgetsAction = {
-      handler: function() {
-        delete Jupyter.notebook.metadata.widgets;
-        Jupyter.menubar.actions.get('jupyter-notebook:save-notebook').handler({
-          notebook: Jupyter.notebook
-        });
-      },
-      help: 'Clear the widget state information from the notebook'
-    };
-    Jupyter.menubar.actions.register(
-      this.saveWidgetsAction,
-      'save-clear-widgets',
-      'widgets'
-    );
+    this.notebook.events.on('before_save.Notebook', () => {
```

Comment:
I like how much cleaner this is... *but* I wonder if this suggests the current save `'save-with-widgets'` should be replaced with a `'save-without-widgets'` option? Or alternatively, just maintain `'save-clear-widgets'` and bind that to the old menu item (line 288 before this PR).

That way the default would be to save (which I think is more intuitive), but there's still a way to save without the widget state for the cases where the widget really needs to be saved that way, and an advanced user can rebind keys to do that etc.